### PR TITLE
[CodeCompletion] Infer keypath root type from the context type

### PR DESCRIPTION
**Pull request by rintaro**
So that this works:

```swift
func foo(_: KeyPath<SomeObject, Int>) {}
foo(\.<HERE>)
```

rdar://problem/46102807

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: lib/IDE/CodeCompletion.cpp [hunk 1]
```diff
@@ -5004,19 +5004,30 @@ void CodeCompletionCallbacksImpl::doneParsing() {
     bool OnRoot = !KPE->getComponents().front().isValid();
     Lookup.setIsSwiftKeyPathExpr(OnRoot);
 
-    auto ParsedType = BGT->getGenericArgs()[1];
-    auto Components = KPE->getComponents();
-    if (Components.back().getKind() ==
-        KeyPathExpr::Component::Kind::OptionalWrap) {
+    Type baseType = BGT->getGenericArgs()[OnRoot ? 0 : 1];
+    if (OnRoot && baseType->is<UnresolvedType>()) {
+      // Infer the root type of the keypat from the context type.
```
Comment (benlangmuir) [inline lib/IDE/CodeCompletion.cpp]: typo: keypat -> keypath

**Review by benlangmuir — APPROVED** [review]
LGTM; one typo in comment.

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Pull request merged**